test/unittest/dds/xtypes/type_representation/TypeObjectUtilsTests.cpp

```diff
@@ -461,93 +495,398 @@ TEST(TypeObjectUtilsTests, build_plain_map_s_type_defn_inconsistencies)
     EXPECT_NO_THROW(key_identifier->_d(TK_INT64));
-    EXPECT_NO_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
                 fully_descriptive_header, 10, test_identifier, flags, key_identifier));
     StringSTypeDefn string_type_def = TypeObjectUtils::build_string_s_type_defn(50);
     EXPECT_NO_THROW(key_identifier->string_sdefn(string_type_def));
-    EXPECT_NO_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
+    EXPECT_NO_THROW(PlainMapSTypeDefn plain_map = TypeObjectUtils::build_plain_map_s_type_defn(
                 fully_descriptive_header, 10, test_identifier, flags, key_identifier));
-    // Change discriminator to EK_COMPLETE
+
+    // Get EK_COMPLETE and EK_MINIMAL TypeIdentifiers
     register_empty_structure_type_object();
     TypeIdentifierPair type_ids;
+    eprosima::fastcdr::external<TypeIdentifier> complete_typeid;
+    eprosima::fastcdr::external<TypeIdentifier> minimal_typeid;
     DomainParticipantFactory::get_instance()->type_object_registry().get_type_identifiers("empty_structure", type_ids);
     if (EK_COMPLETE == type_ids.type_identifier1()._d())
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
+        complete_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
+        minimal_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
     }
     else
     {
-        test_identifier = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        complete_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier2()));
+        minimal_typeid = eprosima::fastcdr::external<TypeIdentifier>(new TypeIdentifier(type_ids.type_identifier1()));
     }
-    // TypeIdentifier consistent with complete header
-    EXPECT_NO_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                complete_header, 10, test_identifier, flags, key_identifier));
-    // TypeIdentifier inconsistent with minimal header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                minimal_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
-    // TypeIdentifier inconsistent with fully-descriptive header
-    EXPECT_THROW(PlainMapSTypeDefn plain_seq = TypeObjectUtils::build_plain_map_s_type_defn(
-                fully_descriptive_header, 10, test_identifier, flags, key_identifier), InvalidArgumentError);
-    // Change discriminator to EK_MINIMAL
-    if (EK_COMPLETE == type_ids.type_identifier1()._d())
+
+    register_plain_seq_type_object(complete_typeid);
+    TypeIdentifierPair indirect_type_ids;
```

Comment:
Why not use the already defined variable `type_ids`? The information have already been copied into the variables `complete_typeid` and `minimal_typeid`.

Reply:
reused for all cases in [d3245e1](https://github.com/eProsima/Fast-DDS/pull/4496/commits/d3245e16dd687c62ce397be8018c718828d2bc6e)